identity_jose/tests/custom_alg.rs

```diff
@@ -0,0 +1,102 @@
+use std::ops::Deref;
+use std::time::SystemTime;
+
+use crypto::signatures::ed25519::{PublicKey, SecretKey, Signature};
+use identity_jose::jwk::JwkParamsOkp;
+use identity_jose::jwk::JwkType;
+use identity_jose::jwk::{EdCurve, Jwk};
+use identity_jose::jws::CompactJwsEncoder;
+use identity_jose::jws::Decoder;
+use identity_jose::jws::JwsAlgorithm;
+use identity_jose::jws::JwsVerifierFn;
+use identity_jose::jws::VerificationInput;
+use identity_jose::jws::{JwsHeader, SignatureVerificationError, SignatureVerificationErrorKind};
+use identity_jose::jwt::JwtClaims;
+use identity_jose::jwu;
+use jsonprooftoken::encoding::base64url_decode;
+
+#[test]
+fn custom_alg_roundtrip() {
```

Comment:
@itsyaasir: What do you mean by more specialized? The test that I added already sets a custom `alg` value, ensures it's part of the serialized JSON and that the custom `alg` value is passed to the `JwsVerifier`.

I don't think there is any more places where the custom `alg` value would be relevant. 

To actually use the signature scheme that we need the feature for (ICP canister signatures), I would need to pull in additional dependencies (which I wanted to avoid).
But I can add a test that uses it of course, if you think it is useful.